sympy/printing/pretty/pretty.py

```diff
@@ -1902,7 +1902,10 @@ def _print_Mul(self, product):
             return prettyForm.__mul__(*a)/prettyForm.__mul__(*b)
 
     # A helper function for _print_Pow to print x**(1/n)
-    def _print_nth_root(self, base, expt):
+    def _print_nth_root(self, base, expt, den_p):
```

Comment:
Why does this function need `exp` at all? Shouldn't we just pass in `root`, and then compute `broot = self._print(root)` inside the function and use it instead of `den_p`?

Reply:
I don't fully understand what you are trying to say.  The function can be made to work instead of `expt` and the current if-else block, yes, and I will make that change, but what do you mean by `root` variable? Are you referring to the denominator `d`? Because the denominator needs to be passed.

Reply:
Yes, the denominator `d` is the degree of the root, right? So I'm suggesting passing `d` at the caller into an argument called `root`